packages/instantsearch-react/src/InstantSearch.js

```diff
@@ -1,141 +1,329 @@
-import React, {PropTypes, Component} from 'react';
+import {PropTypes, Component, Children} from 'react';
 import algoliasearch from 'algoliasearch';
 import algoliasearchHelper, {SearchParameters} from 'algoliasearch-helper';
-import {Provider} from 'react-algoliasearch-helper';
-import omit from 'lodash/object/omit';
-import isEqual from 'lodash/lang/isEqual';
-import {createHistory, createMemoryHistory} from 'history';
+import {omit, isEqual, includes} from 'lodash';
+import qs from 'qs';
+import {createHistory} from 'history';
 
-import createConfigManager from './createConfigManager';
-import createStateManager from './createStateManager';
-import {configManagerPropType, stateManagerPropType} from './propTypes';
+import createWidgetsManager from './createWidgetsManager';
+import createStore from './createStore';
+
+function getStateFromLocation(location) {
+  if (location.query) {
+    return location.query;
+  }
+  // We could also use location.query with the useQueries enhancer, but that
+  // would require a bit more configuration from the user.
+  return qs.parse(location.search.slice(1));
+}
+
+function alphabeticalSort(a, b) {
+  if (a > b) {
+    return 1;
+  }
+  if (a === b) {
+    return 0;
+  }
+  return 1;
+}
+
+function applyStateToLocation(location, state, knownKeys) {
+  const urlState = getStateFromLocation(location);
+  const unknownParameters = omit(urlState, knownKeys);
+  const query = {
+    ...unknownParameters,
+    ...state,
+  };
+  if (location.query) {
+    return {
+      ...location,
+      query,
+    };
+  }
+  return {
+    ...location,
+    search: query ? `?${qs.stringify(query, {sort: alphabeticalSort})}` : '',
+  };
+}
+
+function formatProps(props) {
+  return props.map(prop => `\`${prop}\``).join(', ');
```

Comment:
Yep, it is, although ESLint would want us to do
```
return `\`${props.join('`, `')}\``;
```
Although it's a bit longer, I prefer the map/join version, it has clearer intent imho.